From 761ba8edad5e194c07d6829f10c7b71e6e4f7294 Mon Sep 17 00:00:00 2001
From: sgnanase <sundar.gnanasekaran@intel.com>
Date: Thu, 7 Mar 2019 14:08:46 +0530
Subject: [PATCH] [APL-NUC][Q][REVERT-ME] Fix issue of apl_nuc after aosp
 rebase

Stop re-tries of getting HALs whose modules are not loaded
To be reverted once actual fix is merged

Change-Id: I6061bd6b6a26dc0aaefb35953b364000504b3657
Tracked-On: https://jira.devtools.intel.com/browse/OAM-76839
Signed-off-by: sgnanase <sundar.gnanasekaran@intel.com>
---
 transport/ServiceManagement.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/transport/ServiceManagement.cpp b/transport/ServiceManagement.cpp
index e7bec41..ad193cd 100644
--- a/transport/ServiceManagement.cpp
+++ b/transport/ServiceManagement.cpp
@@ -800,7 +800,9 @@ sp<::android::hidl::base::V1_0::IBase> getRawServiceInternal(const std::string&
         }

         // In case of legacy or we were not asked to retry, don't.
-        if (vintfLegacy || !retry) break;
+        // Skip retry for HALs that are not laoded
+        retry=0;
+        if (!retry) break;

         if (waiter != nullptr) {
             ALOGI("getService: Trying again for %s/%s...", descriptor.c_str(), instance.c_str());
--
2.21.0
